File: src/my_project/easy_problems/from201to250/minimum_changes_to_make_alternating_binary_string.py
from typing import List 


class Solution:
    def minOperations(self, s: str) -> int:
        
        first_dict = {}
        second_dict = {}
        len_s: int = len(s)
        min_count_first_dict: int = 0
        min_count_second_dict: int = 0
        
        for i in range(len_s):
            first_dict[i] = str((i+1)%2)
            
            
        for i in range(len_s):
            if s[i] == first_dict[i]:
                min_count_first_dict += 1
            else: 
                min_count_second_dict += 1
                
        answer = min(min_count_first_dict, min_count_second_dict)
                
            
        
        
        return answer 
    
    
solution = Solution()


print(solution.minOperations( s = "0100"))

print(solution.minOperations(s = "10"))

print(solution.minOperations(s = "1111"))

print(solution.minOperations("10010100"))

File: src/my_project_typescript/easy_problems_typescript/from101to150/minimumChangesToMakeAlternatingBinaryString.ts
function minOperations(s: string): number {


    
    let firstDict: {[key:number]: string} = {}
    

    let lenS: number = s.length
    let minCountFirstDict: number = 0
    let minCountSecondDict: number = 0

    for (let i = 0; i < lenS; i++){
        firstDict[i] = ((i+1)%2).toString()
    }

    for (let i = 0; i < lenS; i++){
        if (s[i] === firstDict[i]){
            minCountFirstDict += 1
        }
        else {
            minCountSecondDict += 1
        }
    }

    let answer: number = 
    Math.min(minCountFirstDict,minCountSecondDict)


    return answer 

};

console.log(minOperations("0100"))

console.log(minOperations("10"))

console.log(minOperations("1111"))

console.log(minOperations("10010100"))
